text/0083-func-config-standard.md

```diff
@@ -0,0 +1,66 @@
+- **TEP**: [0](https://github.com/ton-blockchain/TEPs/pull/0)
+- **title**: FunC project configuration file
+- **status**: Draft
+- **type**: Meta
+- **authors**: [Danila Volkov](https://github.com/dvlkv), [Andrey Pfau](https://github.com/andreypfau)
+- **created**: 15.08.2022
+
+# Summary
+
+I propose to make a FunC project config, like project.json, Cargo.toml, etc.
+
+# Motivation
+
+Extensions, compilers and other dev tools need universal project configuration.
+
+# Guide
+
+Extensions can use it for optimal indexing source code, compilers can be configured without bash scripts.  
+Also, here comes more convinient project bootstapping. 
+
+# Specification
+
+I suppose it should be yaml file, named like .func.yml or something else.
+In first implementation it should contain:
+- root source folder
+- entrypoints (contract root files), relative to source folder
+- (optional) library directories
+- (optional) compiler configuration (version, etc.)  
```

Comment:
I think it's must be not optional, just need to add version of compiler for any. While each version can present new syntax.

Reply:
Compiler version can be declared using #pragma, so I suppose that it is just an alternative to it.

Reply:
got it. But my idea is to force all devs to write down version of compiler. It can be that you found project and can't compile it because it miss-versioned. I have encountered this very often. This little action will make life very easy.